Update TraderX samples inline with the C4. Part of response to #47

rocketstack-matt opened the pull request:
Doesn't address external node question).

@Budlee / @aidanm3341 I think these changes should rectify the issues you were both having. Would be great to see the updated visualisations in case I've missed something.

Also, I've duplicated some relationships to match the C4 design, although suspect we may need to think about how we may want to address this without the duplication.

aidanm3341 commented:
Plugged it in as is and the new relationships cause a lot of noise and aren't clear without the descriptions. I could try adding in the descriptions, making it more C4-esque.
<img width="1482" alt="image" src="https://github.com/finos-labs/architecture-as-code/assets/7470072/168702c9-22b3-41e4-abf7-4693d2dc4b7b">

rocketstack-matt commented:
Descriptions weren't a thing in the schema, but I added that in PR #50 and added the actual descriptions in this PR if you look at the JSON / YAML files.

Budlee commented:
Initially looks better from what I can see. I'd need to play around a bit more but this has really cleared up for me what I should expect in the json 

```mermaid
graph TB
  linkStyle default fill:#ffffff

  subgraph diagram ["TraderX - Containers"]
    style diagram fill:#ffffff,stroke:#ffffff

    12["<div style='font-weight: bold'>Trader</div><div style='font-size: 70%; margin-top: 0px'>[Person]</div><div style='font-size: 80%; margin-top:10px'>Person who manages accounts<br />and executes trades</div>"]
    style 12 fill:#dddddd,stroke:#9a9a9a,color:#000000

    subgraph 1 [TraderX]
      style 1 fill:#ffffff,stroke:#9a9a9a,color:#9a9a9a

      10["<div style='font-weight: bold'>Trade Processor</div><div style='font-size: 70%; margin-top: 0px'>[Container: service]</div><div style='font-size: 80%; margin-top:10px'>Process incoming trade<br />requests, settle and persist</div>"]
      style 10 fill:#dddddd,stroke:#9a9a9a,color:#000000
      11["<div style='font-weight: bold'>Accounts Service</div><div style='font-size: 70%; margin-top: 0px'>[Container: service]</div><div style='font-size: 80%; margin-top:10px'>Service which provides<br />account management</div>"]
      style 11 fill:#dddddd,stroke:#9a9a9a,color:#000000
      2["<div style='font-weight: bold'>Web Client</div><div style='font-size: 70%; margin-top: 0px'>[Container: webclient]</div><div style='font-size: 80%; margin-top:10px'>Browser based web interface<br />for TraderX</div>"]
      style 2 fill:#dddddd,stroke:#9a9a9a,color:#000000
      26["<div style='font-weight: bold'>User Directory</div><div style='font-size: 70%; margin-top: 0px'>[Container: ldap]</div><div style='font-size: 80%; margin-top:10px'>Golden source of user data</div>"]
      style 26 fill:#dddddd,stroke:#9a9a9a,color:#000000
      3["<div style='font-weight: bold'>Web GUI</div><div style='font-size: 70%; margin-top: 0px'>[Container: service]</div><div style='font-size: 80%; margin-top:10px'>Allows employees to manage<br />accounts and book trades</div>"]
      style 3 fill:#dddddd,stroke:#9a9a9a,color:#000000
      4["<div style='font-weight: bold'>Position Service</div><div style='font-size: 70%; margin-top: 0px'>[Container: service]</div><div style='font-size: 80%; margin-top:10px'>Server process which<br />processes trading activity<br />and updates positions</div>"]
      style 4 fill:#dddddd,stroke:#9a9a9a,color:#000000
      5["<div style='font-weight: bold'>TraderX DB</div><div style='font-size: 70%; margin-top: 0px'>[Container: database]</div><div style='font-size: 80%; margin-top:10px'>Database which stores<br />account, trade and position<br />state</div>"]
      style 5 fill:#dddddd,stroke:#9a9a9a,color:#000000
      6["<div style='font-weight: bold'>People Service</div><div style='font-size: 70%; margin-top: 0px'>[Container: service]</div><div style='font-size: 80%; margin-top:10px'>Service which provides user<br />details management</div>"]
      style 6 fill:#dddddd,stroke:#9a9a9a,color:#000000
      7["<div style='font-weight: bold'>Reference Data Service</div><div style='font-size: 70%; margin-top: 0px'>[Container: service]</div><div style='font-size: 80%; margin-top:10px'>Service which provides<br />reference data</div>"]
      style 7 fill:#dddddd,stroke:#9a9a9a,color:#000000
      8["<div style='font-weight: bold'>Trading Services</div><div style='font-size: 70%; margin-top: 0px'>[Container: service]</div><div style='font-size: 80%; margin-top:10px'>Service which provides<br />trading services</div>"]
      style 8 fill:#dddddd,stroke:#9a9a9a,color:#000000
      9["<div style='font-weight: bold'>Trade Feed</div><div style='font-size: 70%; margin-top: 0px'>[Container: service]</div><div style='font-size: 80%; margin-top:10px'>Message bus for streaming<br />updates to trades and<br />positions</div>"]
      style 9 fill:#dddddd,stroke:#9a9a9a,color:#000000
    end

    12-. "<div></div><div style='font-size: 70%'></div>" .->2
    2-. "<div>Web client interacts with the<br />Web GUI process.</div><div style='font-size: 70%'>[HTTPS]</div>" .->3
    3-. "<div>Load positions for account.</div><div style='font-size: 70%'>[HTTPS]</div>" .->4
    3-. "<div>Load trades for account.</div><div style='font-size: 70%'>[HTTPS]</div>" .->4
    4-. "<div>Looks up default positions<br />for a given account.</div><div style='font-size: 70%'>[JDBC]</div>" .->5
    4-. "<div>Looks up all trades for a<br />given account.</div><div style='font-size: 70%'>[JDBC]</div>" .->5
    3-. "<div>Looks up securities to assist<br />with creating a trade ticket.</div><div style='font-size: 70%'>[HTTPS]</div>" .->7
    3-. "<div>Creates new trades and<br />cancels existing trades.</div><div style='font-size: 70%'>[HTTPS]</div>" .->8
    3-. "<div>Subscribes to trade/position<br />updates feed for currently<br />viewed account.</div><div style='font-size: 70%'>[WebSocket]</div>" .->9
    10-. "<div>Processes incoming trade<br />requests, persist and publish<br />updates.</div><div style='font-size: 70%'>[SocketIO]</div>" .->9
    10-. "<div>Looks up current positions<br />when bootstrapping state,<br />persist trade state and<br />position state.</div><div style='font-size: 70%'>[JDBC]</div>" .->5
    3-. "<div>Creates/Updates accounts.<br />Gets list of accounts.</div><div style='font-size: 70%'>[HTTPS]</div>" .->11
    3-. "<div>Looks up people data based on<br />typeahead from GUI.</div><div style='font-size: 70%'>[HTTPS]</div>" .->6
    6-. "<div>Looks up people data.</div><div style='font-size: 70%'>[LDAP]</div>" .->26
    8-. "<div>Validates securities when<br />creating trades.</div><div style='font-size: 70%'></div>" .->7
    8-. "<div>Publishes updates to trades<br />and positions after<br />persisting in the DB.</div><div style='font-size: 70%'></div>" .->9
    8-. "<div>Validates accounts when<br />creating trades.</div><div style='font-size: 70%'></div>" .->11
  end
```
